core/src/errors.rs

```diff
@@ -184,11 +185,18 @@ impl convert::From<Box<dyn Any + Send>> for CometError {
         }
     }
 }
-
+impl From<DataFusionError> for CometError {
+    fn from(value: DataFusionError) -> Self {
+        CometError::DataFusion {
+            msg: value.message().to_string(),
+            source: value,
+        }
+    }
+}
```

Comment:
nit: looks like we need some blank lines to separate this block

```suggestion

impl From<DataFusionError> for CometError {
    fn from(value: DataFusionError) -> Self {
        CometError::DataFusion {
            msg: value.message().to_string(),
            source: value,
        }
    }
}

```